### PR TITLE
CP-43799: Add a flag to show if an update sync is in progress

### PR DESCRIPTION
**Pull request by gangj**
Set "sync_with_yum_repos_in_progress" as true in "Pool.other_config" when update sync starts, set it to false when it finishes.

### CODE REVIEW

**Review by lindig — APPROVED** [review]

File: ocaml/xapi/xapi_pool.ml [hunk 1]
```diff
@@ -3401,32 +3401,42 @@ let sync_updates ~__context ~self ~force ~token ~token_id =
...
-      @@ fun () ->
-      List.iter (fun x -> create_pool_repository ~__context ~self:x) enabled ;
-      let checksum = set_available_updates ~__context in
-      Db.Pool.set_last_update_sync ~__context ~self ~value:(Date.now ()) ;
-      checksum
+  finally
+    (fun () ->
+      Db.Pool.add_to_other_config ~__context ~self
+        ~key:"sync_with_yum_repo_in_progress" ~value:"true" ;
+      let enabled = Repository_helpers.get_enabled_repositories ~__context in
```
[the remark this answers is not in the document]
Comment (gangj) [reply ocaml/xapi/xapi_pool.ml:3408]: renamed to sync_with_yum_repos_in_progress

File: ocaml/xapi/xapi_pool.ml [hunk 2]
```diff
@@ -3401,32 +3401,42 @@ let sync_updates ~__context ~self ~force ~token ~token_id =
...
-      checksum
+  finally
+    (fun () ->
+      Db.Pool.add_to_other_config ~__context ~self
+        ~key:"sync_with_yum_repos_in_progress" ~value:"true" ;
+      let enabled = Repository_helpers.get_enabled_repositories ~__context in
+      match force with
+      | true ->
+          Xapi_pool_helpers.with_pool_operation ~__context ~self
+            ~doc:"pool.sync_updates" ~op:`sync_updates
```
[the remark this answers is not in the document]
Comment (gangj) [reply ocaml/xapi/xapi_pool.ml:3412]: > I just realized this: could the client see this `op` in `pool.current_operations`?

OK, will try with jsonrpc api call to get pool record to have a check.

[the remark this answers is not in the document]
Comment (gangj) [reply ocaml/xapi/xapi_pool.ml:3412]: OK, will try with jsonrpc api call to get pool record to have a check.

[the remark this answers is not in the document]
Comment (gangj) [reply ocaml/xapi/xapi_pool.ml:3412]: Setup a pool with toolstack-next build, and a python program which calls XAPI API "pool.get_record" during an update sync, execute the python program several times during the update sync, the update sync takes a long time to finish as there are lots of update data synced:
```
[root@purley-2s ~]# ls /var/lib/yum* ; du -sh /var/lib/yum*
/var/lib/yum:
groups  history  repos  rpmdb-indexes  uuid  yumdb

/var/lib/yum-mirror:
remote-2eebc3d6-306c-b90a-f3db-aaba6fff018b
9.1M    /var/lib/yum
391M    /var/lib/yum-mirror
```

But the `current_operations` value from xapi response is always empty:
```
'current_operations': {}
```

[the remark this answers is not in the document]
Comment (minglumlu) [reply ocaml/xapi/xapi_pool.ml:3412]: That may be another issue.

**Review by minglumlu — APPROVED** [review]

**Comment by robhoes** [comment]
Doing this with `other_config` is not great and we have better, standard ways of handling this kind of thing with `allowed_operations`.

**Comment by gangj** [comment]
> Doing this with `other_config` is not great and we have better, standard ways of handling this kind of thing with `allowed_operations`.

@robhoes thanks~
Test during a pool sync, check the field with a XAPI API call: "pool.get_record":
```
'allowed_operations': [], 'current_operations': {'OpaqueRef:fe97e620-b2a8-ead7-419b-3c8f8f1784c3': 'sync_updates'},
```
Will check with Tina about it.

**Comment by robhoes** [comment]
XC has these fields in its cache, and uses them all the time.

**Pull request merged**